client/checkout/blocks/upe-split.js

```diff
@@ -71,22 +72,32 @@ Object.entries( enabledPaymentMethodsConfig )
 	.map( ( [ upeName, upeConfig ] ) =>
 		registerPaymentMethod( {
 			name: upeMethods[ upeName ],
-			content: (
-				<WCPayUPEFields
-					paymentMethodId={ upeName }
-					upeMethods={ upeMethods }
-					api={ api }
-					testingInstructions={ upeConfig.testingInstructions }
-				/>
-			),
-			edit: (
-				<WCPayUPEFields
-					paymentMethodId={ upeName }
-					upeMethods={ upeMethods }
-					api={ api }
-					testingInstructions={ upeConfig.testingInstructions }
-				/>
-			),
+			content: getUPEConfig( 'isUPESplitEnabled' )
+				? getSplitUPEFields(
+						upeName,
+						upeMethods,
+						api,
+						upeConfig.testingInstructions
+				  )
+				: getDeferredIntentCreationUPEFields(
+						upeName,
+						upeMethods,
+						api,
+						upeConfig.testingInstructions
+				  ),
```

Comment:
```suggestion
			content: getUPEConfig( 'isUPEDeferredEnabled' )
				? getDeferredIntentCreationUPEFields(
						upeName,
						upeMethods,
						api,
						upeConfig.testingInstructions
				  )
				: getSplitUPEFields(
						upeName,
						upeMethods,
						api,
						upeConfig.testingInstructions
				  ),
```
Thoughts on checking the deferred intent feature flag here instead? The advantage is we can rely on the most relevant feature flag alone and can ensure that on rollout we only have to switch on the new flag and not worry about the previous flag. This would theoretically allow us to distinguish between users who have upgraded to deferred intent UPE from split UPE and those upgrading from previous non-UPE sites (permitting to fall back to the correct feature flag, if we disabled deferred intent UPE for whatever reason).

It's a minor issue and there are good arguments in either direction, so I can leave this one to your discretion. 🤷

Reply:
This topic is an excellent area to refine further, and I'm happy you started the discussion. This script will be executed in three scenarios: 
1. split UPE FF enabled
2. deferred intent UPE FF enabled
3. both FFs enabled

The first two points are irrelevant in our discussion since, regardless of the final approach we'll go with, the proper implementation will be chosen. For the third point, we'd need to decide which implementation takes precedence. 

> The advantage is we can rely on **the most relevant feature flag** alone

From your comment and all the details you mentioned above, I understand it's deferred intent UPE that needs to take precedence. Also, I could consider [the shortcode implementation where deferred is a higher priority](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-upe-checkout.php#L126-L132). 

Thus I'll change the implementation shortly to leverage the `isUPEDeferredEnabled` flag instead. 

UPDATE: changed in a5e8c3af52fed2e2a81e600befd00a64b8e95758